tests/unit/test_generate_umm_var.py

```diff
@@ -0,0 +1,276 @@
+        cls.netcdf4_basename = 'f16_ssmis_20210426v7.nc'
+        cls.netcdf4_url = f'https://example.com/{cls.netcdf4_basename}'
+        cls.rssmif16d_variables = ['atmosphere_cloud_liquid_water_content',
+                                   'atmosphere_water_vapor_content',
+                                   'latitude', 'longitude', 'rainfall_rate',
+                                   'sst_dtime', 'time', 'wind_speed']
+
+    @staticmethod
+    def download_granule_side_effect(granule_link, auth_header, out_directory):
+        """ A helper method that will copy the test file to the temporary
+            directory being used for a specific test.
+
+            Static methods do not have access to class attributes, so the test
+            file path is defined in this method as well as setUpClass.
+
+        """
+        netcdf4_file_path = 'tests/unit/data/f16_ssmis_20210426v7.nc'
+        return copy(netcdf4_file_path, out_directory)
+
+    @patch('varinfo.umm_var.publish_umm_var')
+    @patch('varinfo.cmr_search.GranuleQuery')
+    @patch('varinfo.generate_umm_var.download_granule')
+    def test_generate_collection_umm_var_no_publication(self,
+                                                        mock_download_granule,
+                                                        mock_granule_query,
+                                                        mock_publish_umm_var):
+        """ A request with all the necessary information should succeed.
+            This test is moderately end-to-end, but mocks the full
+            download_granule function for simplicity. That function is tested
+            in detail in test_cmr_search.py.
+
+        """
+        mock_granule_query.return_value.get.return_value = [
+            {
+                'links': [
+                    {
+                        'href': self.netcdf4_url,
+                        'rel': 'http://esipfed.org/ns/fedsearch/1.1/data#'
+                    }
+                ]
+            }
+        ]
+
+        # Add side effect that will copy test file to the temporary directory,
+        # simulating a download.
+        mock_download_granule.side_effect = self.download_granule_side_effect
+
+        # Run the test:
+        generated_umm_var = generate_collection_umm_var(
+            self.collection_concept_id,
+            self.bearer_token_header
+        )
+
+        # Ensure the granule query used expected query parameters
+        mock_granule_query.return_value.parameters.assert_called_once_with(
+            downloadable=True,
+            sort_key='-start_date',
+            concept_id=self.collection_concept_id
+        )
+
+        # Ensure the call to download the granule had correct parameters
+        mock_download_granule.assert_called_once_with(self.netcdf4_url,
+                                                      self.bearer_token_header,
+                                                      out_directory=ANY)
```

Comment:
These `ANY` statements are a bit lazy. The issue is that the temporary directory will change every test run. If I had a bit more time, I'd be tempted to write a custom matcher that ensures the string format is as expected. But that maybe is overkill. 🤷